xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -179,11 +363,304 @@ public void handleResolvedAddress_createLbChannel()
+
+    // first EDS response
+    serverResponseWriter.onNext(edsResponse);
+    verify(lookasideChannelCallback).onWorking();
+    ArgumentCaptor<LoadReportCallback> loadReportCallbackCaptor =
+        ArgumentCaptor.forClass(LoadReportCallback.class);
+    verify(loadReportClient).startLoadReporting(loadReportCallbackCaptor.capture());
+    LoadReportCallback loadReportCallback = loadReportCallbackCaptor.getValue();
+
+    // second EDS response
+    serverResponseWriter.onNext(edsResponse);
+    verify(lookasideChannelCallback, times(1)).onWorking();
+    verify(loadReportClient, times(1)).startLoadReporting(any(LoadReportCallback.class));
+
+    LocalityStore localityStore = Iterables.getOnlyElement(localityStores);
+    verify(localityStore, never()).updateOobMetricsReportInterval(anyLong());
+    loadReportCallback.onReportResponse(1234);
+    verify(localityStore).updateOobMetricsReportInterval(1234);
+
+    verify(lookasideChannelCallback, never()).onError();
+
+    lookasideLb.shutdown();
+  }
+
+  @Test
+  public void handleDropUpdates() {
+    lookasideLb.handleResolvedAddresses(defaultResolvedAddress);
+
+    LocalityStore localityStore = Iterables.getOnlyElement(localityStores);
+    verify(localityStore, never()).updateDropPercentage(
+        ArgumentMatchers.<ImmutableList<DropOverload>>any());
+
+    serverResponseWriter.onNext(edsResponse);
+    verify(localityStore).updateDropPercentage(eq(ImmutableList.<DropOverload>of()));
+
+    ClusterLoadAssignment clusterLoadAssignment = ClusterLoadAssignment.newBuilder()
+        .setPolicy(Policy.newBuilder()
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_1").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.HUNDRED)
+                    .setNumerator(3)
+                    .build())
+                .build())
+
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_2").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.TEN_THOUSAND)
+                    .setNumerator(45)
+                    .build())
+                .build())
+            .addDropOverloads(Policy.DropOverload.newBuilder()
+                .setCategory("cat_3").setDropPercentage(FractionalPercent.newBuilder()
+                    .setDenominator(DenominatorType.MILLION)
+                    .setNumerator(6789)
+                    .build())
+                .build())
+            .build())
+        .build();
+    serverResponseWriter.onNext(
+        DiscoveryResponse.newBuilder()
+            .addResources(Any.pack(clusterLoadAssignment))
+            .setTypeUrl("type.googleapis.com/envoy.api.v2.ClusterLoadAssignment")
+            .build());
+
+    verify(lookasideChannelCallback, never()).onAllDrop();
+    verify(localityStore).updateDropPercentage(ImmutableList.of(
+        new DropOverload("cat_1", 300_00),
```

Comment:
FYI: These numbers do not look elegant. Please be consistent for all numbers you write, either put underscores for all or none (in general I don't). You could choose to not add the underscore, but if you do, please do it elegantly. 

I know this is copied from the old code and these tests would eventually go away, or need to be re-written in another form (with `XdsClient`, you do not need a RPC service involved, just need to build `EndpointUpdate`s and test your load balancing behaviors). So you could choose to not change it now.

Reply:
It's just an arbitrary string in test I don't think it necessary to keep that pattern consistent. It dose not affect readability.